datafusion/core/src/datasource/physical_plan/csv.rs

```diff
@@ -1210,6 +1244,44 @@ mod tests {
         crate::assert_batches_eq!(expected, &result);
     }
 
+    #[tokio::test]
+    async fn test_terminator() {
```

Comment:
Perhaps there also should be added sqllogictests for reading files with customized line terminator (also checking that nothing is broken when terminator used along with newlines_in_values).

Reply:
I got two issues when creating the sqllogictests case:
1. I prepared a test file breaking the line by `CR`. However, it seems that object store doesn't support this by default.
```
External error: query failed: DataFusion error: Object Store error: Generic LocalFileSystem error: Requested range was invalid
```
2. I tried to create an external table with the terminator config like
```
statement ok
CREATE EXTERNAL TABLE stored_table_with_cf_terminator (
col1 TEXT,
col2 TEXT
) STORED AS CSV
LOCATION '../core/tests/data/cr_terminator.csv'
OPTIONS ('format.terminator' '\r', 'format.has_header' 'true');
```
However, there are some issues for parsing the value.
```
External error: statement failed: DataFusion error: Invalid or Unsupported Configuration: Error parsing \r as u8. Non-ASCII string provided
```
It seems that `\r` is parsed to two different ASCII but it is one. 🤔 
I tried to select `\r` directly 
```
> select '\t';
+-----------+
| Utf8("	") |
+-----------+
| 	         |
+-----------+
1 row(s) fetched. 
Elapsed 0.003 seconds.
```
It works fine but it's not valid for the config value. Is there any way to correctly create `\r` character for the config value?

Reply:
Yes, it works only for single character -- probably ConfigField [implementation](https://github.com/apache/datafusion/blob/ac74cd3163e43563807a8c6e8e72bb058cb6f459/datafusion/common/src/config.rs#L983) for u8 should be enhanced, as right now it doesn't accept escape sequences like `\t` `\n` etc.

(same works for the delitmiter -- I was'n able to read tab-separated file via SQL :thinking: )

Reply:
To solve the ConfigField issue, I support `EscapedStringLiteral` in 8a1c9c7. Now, we can use `E'\r'` to create the terminator. I think it also works for `E'\t'` or `E'\n'`.